lib/request.js

```diff
@@ -120,7 +120,8 @@ function buildRequest (opts) {
       hostname: opts.url.hostname,
       headers: opts.headers,
       agent: agents[opts.url.protocol.replace(/^unix:/, '')],
-      ...httpOpts.requestOptions
+      ...httpOpts.requestOptions,
+      timeout: opts.timeout || httpOpts.requestOptions.timeout
```

Comment:
Instead of `||`, using `??` is better in this case.
Since timeout can be `0`.

Reply:
@climba03003 thanks for the review! 

Good point